app/controllers/styleguide_controller.rb

```diff
@@ -23,4 +30,41 @@ def components
   def links
     @links = %w[Layout Typography Links Lists Colours]
   end
+
+  def primary_colours
```

Comment:
Could these methods be made into constants instead, please?

Reply:
tbh, I dont think these `constants` belong in the controller, whether as `constants` or `private methods`. I'm not sure of the best place to put them and am conscious that we don't want to spend alot of time on this 'feature'. Any suggestions, maybe in a styleguide_helper? wdyt @tomas-stefano @willhall88 ?

Reply:
Agree about this contants does not belong to the controller. Constants in a styleguide_helper.rb sounds reasonable to me. 👍

Reply:
All these colours are defined in this file https://github.com/moneyadviceservice/fin_cap/blob/master/app/assets/stylesheets/lib/_variables.scss
I don't know whether we can use that file for the variables in the view. If we can then it would keep the styleguide maintenance a lot easier. I have visions of at some point the colours being updated or changed the the styleguide not being updated to reflect this

Reply:
but, as you say, it's not something we want to spend a lot of time on 'fixing' necessarily

Reply:
@margOnline this is what I was referring to. _See below_

```
# file: app/helpers/styleguide_helper.rb 

module StyleguideHelper
  PRIMARY_COLOURS = {
    'yellow'     => '#FFCC00',
    'orange'     => '#EF7E25',
    'blue'       => '#0066CC',
    'blue-dark'  => '#003399',
    'green-pale' => '#51B9AF',
    'purple'     => '#663399'
  }.freeze

  SUPPORTING_COLOURS = {
    'blue' => '#41A0DA',
    'red' => '#C92628',
    'red-pale'     => '#E95E50',
    'green-dark'   => '#008996',
    'green-lime'   => '#8CC635',
    'blue-lighter' => '#e5ebf5'
  }.freeze

  MONO_COLOURS = {
    'black' => '#1D1D1B',
    'white' => '#FFFFFF'
  }.freeze

  GREYS = {
    'light'    => '#DADADA',
    'lighter'  => '#EBE9E9',
    'lightest' => '#F1F1F1'
  }.freeze

  COMPONENTS = {
    'Teaser box'   => 'teaser',
    'Hero image'   => 'hero',
    'Bordered box' => 'borderedbox',
    'Latest news'  => 'latestnews',
    'Download box' => 'downloads',
    'Feedback'     => 'feedback',
    'Large call to action' => 'largecta'
  }.freeze

  LINKS = %w[Layout Typography Links Lists Colours].freeze
end
```

```
#file: app/controllers/styleguide_controller.rb

class StyleguideController < ApplicationController
  before_action :components, :links
  layout 'styleguide'
  include StyleguideHelper

  def show
    if params[:page] == 'colours'
      @primary_colours    = PRIMARY_COLOURS
      @supporting_colours = SUPPORTING_COLOURS
      @mono_colours       = MONO_COLOURS
      @greys              = GREYS
    end

    render params[:page]
  end

  private

  def components
    @components = COMPONENTS
  end

  def links
    @links = LINKS
  end
end
```

Reply:
ok, thx @Tr4pSt3R, all done!